packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -329,6 +338,42 @@ function WorklistDataGrid(props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [dataGridColumns]);
 
+  // useEffect for row updates
+  useEffect(() => {
+    const previousSelectedRows = [...selectedRows.current];
+    selectedRows.current = rows.filter((row) => row.isSelected).map(row => (row.id));
+
+    if (previousSelectedRows.length > 0 && selectedRows.current.length === 0) {
+      setRowSelectionAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.all-rows-unselected' }));
+    } else if (selectedRows.current.length === rows.length) {
+      setRowSelectionAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.all-rows-selected' }));
+    } else {
+      const rowSelectionsAdded = selectedRows.current.filter(row => !previousSelectedRows.includes(row));
+      const rowSelectionsRemoved = previousSelectedRows.filter(row => !selectedRows.current.includes(row));
+      let selectionUpdateAriaMessage = '';
+
+      if (rowSelectionsAdded.length === 1) {
+        const newRowIndex = rows.findIndex(row => row.id === rowSelectionsAdded[0]);
+        const selectedRowLabel = rows[newRowIndex].ariaLabel || newRowIndex + 1;
+        selectionUpdateAriaMessage = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-template' }, { row: selectedRowLabel });
+      } else if (rowSelectionsAdded.length > 1) {
+        selectionUpdateAriaMessage = intl.formatMessage({ id: 'Terra.worklist-data-grid.multiple-rows-selected' }, { rowCount: rowSelectionsAdded.length });
+      }
+
+      if (rowSelectionsRemoved.length === 1) {
+        const removedRowIndex = rows.findIndex(row => row.id === rowSelectionsRemoved[0]);
+        const unselectedRowLabel = rows[removedRowIndex].ariaLabel || removedRowIndex + 1;
+        selectionUpdateAriaMessage += intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-cleared-template' }, { row: unselectedRowLabel });
+      } else if (rowSelectionsRemoved.length > 1) {
+        selectionUpdateAriaMessage += intl.formatMessage({ id: 'Terra.worklist-data-grid.multiple-rows-unselected' }, { rowCount: rowSelectionsRemoved.length });
+      }
+
+      if (selectionUpdateAriaMessage) {
+        setRowSelectionAriaLiveMessage(selectionUpdateAriaMessage);
+      }
```

Comment:
```suggestion
      setRowSelectionAriaLiveMessage(selectionUpdateAriaMessage);
```

Nit: Could we do without the `if` statement here? From what I understand is going on, if `rowSelectionsAdded >= 1` or `rowSelectionsRemoved >=1` then this value changes. 

Maybe if `VisuallyHiddenText` has a different behavior for `null` vs `''` so if that's true ignore. I just tend to not like nested if statements more than what's necessary.